src/components/screens/Connect.tsx

```diff
@@ -67,10 +72,27 @@ export function Connect({ mode }: Props) {
   const [errorValue, setErrorValue] = useState<string | undefined>(undefined);
   const [scanner, setScanner] = useState(mode === "scan");
   const analytics = useAnalytics();
+  const core = useAtomValue(coreAtom);
   const web3wallet = useAtomValue(web3walletAtom);
   const pendingProposals = usePendingProposals(web3wallet);
   const sessions = useSessions(web3wallet);
   const showBackButton = pendingProposals.data.length || sessions.data.length;
+  const [_connectionStatus, setConnectionStatus] = useAtom(
+    relayerConnectionStatusAtom,
+  );
+
+  useEffect(() => {
+    core.relayer.on("relayer_connect", () => {
+      setConnectionStatus("connected");
+    });
+
+    core.relayer.on("relayer_disconnect", () => {
+      setConnectionStatus("disconnected");
+      core.relayer.restartTransport().catch(() => {
+        console.error("couldn't restart transport");
+      });
+    });
+  }, [core]);
```

Comment:
We also need a cleanup function
```suggestion
  useEffect(() => {
    const onConnect = () => {
      setConnectionStatus("connected");
    }
    
    const onDisconnect = () => {
      setConnectionStatus("disconnected");
      core.relayer.restartTransport().catch(() => {
        console.error("couldn't restart transport");
      });
    }
  
    core.relayer.on("relayer_connect", onConnect);
    core.relayer.on("relayer_disconnect", onDisconnect);

    return () => {
      core.relayer.off("relayer_connect", onConnect);
      core.relayer.off("relayer_disconnect", onDisconnect);
    };
  }, [core]);
```

Reply:
And it would probably be good to put this in its own hook function and file

Reply:
Also we might want to put this hook higher in the component hierarchy to make sure we always reconnect